src/Rules/RuleLevelHelper.php

```diff
@@ -221,23 +221,26 @@ public function findTypeToCheck(
 			return new FoundTypeResult(new ErrorType(), [], $errors, null);
 		}
 
-		if (!$this->checkUnionTypes) {
-			if ($type instanceof ObjectWithoutClassType) {
-				return new FoundTypeResult(new ErrorType(), [], [], null);
-			}
-			if ($type instanceof UnionType) {
-				$newTypes = [];
-				foreach ($type->getTypes() as $innerType) {
-					if (!$unionTypeCriteriaCallback($innerType)) {
-						continue;
-					}
-
-					$newTypes[] = $innerType;
-				}
+		if (!$this->checkUnionTypes && $type instanceof ObjectWithoutClassType) {
+			return new FoundTypeResult(new ErrorType(), [], [], null);
+		}
 
-				if (count($newTypes) > 0) {
-					return new FoundTypeResult(TypeCombinator::union(...$newTypes), $directClassNames, [], null);
+		if (
+			!$this->checkUnionTypes && $type instanceof UnionType && !$type instanceof BenevolentUnionType
+			|| !$this->checkBenevolentUnionTypes && $type instanceof BenevolentUnionType
```

Comment:
This needs parentheses for disambiguation, I have no idea what it does now.

Reply:
You mean
```
(
     !$this->checkUnionTypes
     && $type instanceof UnionType
     && !$type instanceof BenevolentUnionType
) || (
    !$this->checkBenevolentUnionTypes
    && $type instanceof BenevolentUnionType
)
```

Or is it more clear another way, like
```
if ($type instanceof UnionType) {
     if (
             (!$this->checkUnionTypes && !$type instanceof BenevolentUnionType)
              || (!$this->checkBenevolentUnionTypes && $type instanceof BenevolentUnionType)
         ) {
         ...
         }
}
```

What do you prefer ?

Reply:
This:

```php
(
	!$this->checkUnionTypes
	&& $type instanceof UnionType
	&& !$type instanceof BenevolentUnionType
) || (
	!$this->checkBenevolentUnionTypes
	&& $type instanceof BenevolentUnionType
)
```